docs/src/index.md

```diff
@@ -268,6 +268,222 @@ df2 = @eachrow df begin
 end
 ```
 
+## Row-wise transformations with `@byrow`
+
+DataFrames.jl provides the function-wrapper `ByRow`. `ByRow(f)(x, y)`
+is roughly equivalent to `f.(x, y)`, with a few exceptions discussed below. 
+DataFramesMeta.jl allows users to construct expressions using `ByRow` 
+function wrapper with the syntax `@byrow`. 
+
+`@byrow` is not a "real" macro and cannot be used outside of 
+DataFramesMeta macros. However it's behavior within DataFramesMeta.jl
+macros should be indistinguishable from externally defined macros. 
+Thought of as a macro `@byrow` accepts a single argument and 
+creates an anonymous function wrapped in `ByRow`.  For example,
+
+```julia
+@transform(df, y = @byrow :x == 1 ? "true" : "false)
+```
+
+becomes
+
+```julia
+transform(df, :x => ByRow(x -> x == 1 ? "true", "false") => :y)
+```
+
+Macros that accept `@byrow`:
+
+* `@transform` and `@transform!`, `@select`, `@select!`, and `@combine`. `
+  @byrow` appears in the right hand side of expressions, of the form `@select(
+  df, z = @byrow :x * :y)`. 
+* `@where` and `@orderby`, with syntax of the form `@where(df, @byrow :x > :y)`
+* `@with`, where the anonymous function created by `@with` is wrapped in
+  `ByRow`. `@with df @byrow :x * :y`, which is conceptually similar to 
+  `map((x, y) -> x * y, df.x, df.y)`.
+
+To avoid writing `@byrow` multiple times when performing multiple
+operations, it is allowed `@byrow` at the beginning of a block of 
+operations. All transformations in the block will operate by row.
+
+```julia
+julia> @where df @byrow begin 
+           :a > 1
+           :b < 5
+       end
+1×2 DataFrame
+ Row │ a      b     
+     │ Int64  Int64 
+─────┼──────────────
+   1 │     2      4
+```
+
+`@byrow` can be used on macros which accept `GroupedDataFrame`s,
+however, like with `ByRow` in DataFrames.jl, when `@byrow` is
+used, functions do not take advantage of the grouping, so the 
+behavior of `@transform(df, y = @byrow f(:x))` and 
+`@transform(groupby(df, :g), y = @byrow f(:x))` is the same.
+
+### Comparison with `@eachrow`
+
+In previous versions of DataFramesMeta, `@eachrow` was named `@byrow`. 
+The old macro `@byrow` is deprecated, but the `@byrow` syntac can 
```

Comment:
```suggestion
The old macro `@byrow` is deprecated, but the `@byrow` syntax can 
```

Reply:
No. By "`@byrow` syntax" I mean `@with df @byrow ...` is very similar to `@eachrow df ...`. In that it's looking through rows.